lib/src/main/cpp/include/surf_detector.h

```diff
@@ -0,0 +1,21 @@
+#ifndef FEATURELIB_INCLUDE_SURF_DETECTOR_H_
+#define FEATURELIB_INCLUDE_SURF_DETECTOR_H_
+
+#include <opencv2/xfeatures2d.hpp>
+#include "feature_detector.h"
+
+namespace featurelib {
+
+class SurfDetector : public FeatureDetector {
+ public:
+  SurfDetector(int width, int height);
+
+  std::shared_ptr<DetectionResult> detect(const std::vector<std::uint8_t> &input) const override;
```

Comment:
whether std::shared_ptr defined in header <memory>? If it's true. i can't find corresponding include

Reply:
Nope, it's a transitive dependency: `#include <utility>` in `detection_result.h` -> `feature_detector.h` -> this file. That's not great, you're right.